homeassistant/components/zwave_me/sensor.py

```diff
@@ -0,0 +1,114 @@
+"""Representation of a sensorMultilevel."""
+from __future__ import annotations
+
+from zwave_me_ws import ZWaveMeData
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    LIGHT_LUX,
+    POWER_WATT,
+    SIGNAL_STRENGTH_DECIBELS,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ZWaveMeController, ZWaveMeEntity
+from .const import DOMAIN
+
+SENSORS_MAP: dict[str, SensorEntityDescription] = {
+    "meterElectric_watt": SensorEntityDescription(
+        key="meterElectric_watt",
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=POWER_WATT,
+    ),
+    "meterElectric_kilowatt_hour": SensorEntityDescription(
+        key="meterElectric_kilowatt_hour",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    "meterElectric_voltage": SensorEntityDescription(
+        key="meterElectric_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+    ),
+    "light": SensorEntityDescription(
+        key="light",
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=LIGHT_LUX,
+    ),
+    "noise": SensorEntityDescription(
+        key="noise",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+    ),
+    "currentTemperature": SensorEntityDescription(
+        key="currentTemperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    "temperature": SensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    "generic": SensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
```

Comment:
Should the generic sensor description really be a temperature sensor?

Reply:
What class should we use for a generic sensor?

Reply:
Probably no class at all. Setting a device class means we know that it's a certain measurement kind.